spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-specialist-link-tracker.spec.js

```diff
@@ -715,4 +715,75 @@ describe('A specialist link tracker', function () {
       }
     })
   })
+
+  describe('when data-ga4-link and data-ga4-limit-to-element-class are on the parent', function () {
+    var specialistLinkTracker
+
+    beforeEach(function () {
+      window.dataLayer = []
+      links = document.createElement('div')
+      links.setAttribute('data-module', 'ga4-link-tracker')
+      links.setAttribute('data-ga4-track-links-only', '')
+      links.setAttribute('data-ga4-limit-to-element-class', 'hello')
+      links.setAttribute('data-ga4-link', JSON.stringify({ event_name: 'navigation', type: 'callout' }))
+
+      links.innerHTML = '<div class="hello"><a class="otherLink" href="https://example.com">GA4 Link Tracker</a></div>' +
+      '<a class="specialistLink" href="https://example.co.uk">Specialist Link Tracker</a>'
```

Comment:
Suggestion: this would be more readable on multiple lines, e.g.

```JS
links.innerHTML = 
  '<div class="hello">' +
    '<a href="https://example.com">GA4 Link Tracker</a>' +
  '</div>' +
  '<a href="https://example.co.uk">Specialist Link Tracker</a>'
```